src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -202,6 +202,8 @@ public abstract class Index : ITombstoneAware, IDisposable, ILowMemoryHandler
 
         private static readonly Size DefaultMaximumMemoryAllocation = new Size(32, SizeUnit.Megabytes);
 
+        private static FifoSemaphore _concurrentlyRunningIndexes;
```

Comment:
Not a good idea, we may run into issues with concurrency here because of tests using different ServerStores.
Better to simply set it on the ServerStore directly.

Reply:
Very good point. Moved to `ServerStore`